app/src/main/java/org/jellyfin/androidtv/ui/picture/PictureViewerViewModel.kt

```diff
@@ -0,0 +1,103 @@
+package org.jellyfin.androidtv.ui.picture
+
+import androidx.lifecycle.ViewModel
+import androidx.lifecycle.viewModelScope
+import kotlinx.coroutines.Dispatchers
+import kotlinx.coroutines.Job
+import kotlinx.coroutines.delay
+import kotlinx.coroutines.flow.MutableStateFlow
+import kotlinx.coroutines.flow.asStateFlow
+import kotlinx.coroutines.isActive
+import kotlinx.coroutines.launch
+import org.jellyfin.sdk.api.client.ApiClient
+import org.jellyfin.sdk.api.client.extensions.itemsApi
+import org.jellyfin.sdk.api.client.extensions.userLibraryApi
+import org.jellyfin.sdk.model.api.BaseItemDto
+import org.jellyfin.sdk.model.api.BaseItemKind
+import org.jellyfin.sdk.model.api.ItemFields
+import org.jellyfin.sdk.model.api.SortOrder
+import java.util.UUID
+import kotlin.time.Duration.Companion.seconds
+
+class PictureViewerViewModel(private val api: ApiClient) : ViewModel() {
+	private var album: List<BaseItemDto> = emptyList()
+	private var albumIndex = -1
+
+	private val _currentItem = MutableStateFlow<BaseItemDto?>(null)
+	val currentItem = _currentItem.asStateFlow()
+
+	suspend fun loadItem(id: UUID) {
+		// Load requested item
+		val itemResponse by api.userLibraryApi.getItem(itemId = id)
+		_currentItem.value = itemResponse
+
+		val albumResponse by api.itemsApi.getItemsByUserId(
+			parentId = itemResponse.albumId,
```

Comment:
Album refers to music albums, so I think you would want to do something like the following:

```suggestion
			parentId = if (itemResponse.type === BaseItemKind.PHOTO_ALBUM) itemResponse.id else itemResponse.parentId,
```

Reply:
In the createIntent() function we check if the given BaseItem is PHOTO so it can never by PHOTO_ALBUM in this place.